clover.rb

```diff
@@ -142,17 +142,11 @@ def self.freeze
     secret: Config.clover_session_secret
 
   autoload_normal("serializers/web", include_first: true)
-
-  # YYY: It'd be nice to use autoload, but it can't work while
-  # constants used across files are defined inside routes files and
-  # the autoload dependency cannot be tracked cheaply.
-  #
-  # if Unreloader.autoload?
-  #   plugin :autoload_hash_branches
-  #   autoload_hash_branch_dir("./routes")
-  # end
-
-  Unreloader.require("routes", delete_hook: proc { |f| hash_branch(File.basename(f).delete_suffix(".rb")) }) {}
+  plugin :autoload_hash_branches
```

Comment:
This is probably fine for now but I wonder what this plugin actually does, and whether we can get around it production mode. I know loader.rb does the "and then force everything to load" pattern, but the routes stuff here is its own system outside loader.rb. Maybe those should be eagerly required in production per Unreloader (and not Zeitwerk) tradition, I don't know.

Reply:
I was expecting to see some Unreloader stuff. But it `require file`

https://github.com/jeremyevans/roda/blob/ba08c961e7a6df92e3f49ec7484b6b199112eca6/lib/roda/plugins/autoload_hash_branches.rb#L43
```ruby
        def autoload_hash_branch(namespace='', segment, file)
          segment = "/#{segment}"
          file = File.expand_path(file)
          opts[:autoload_hash_branch_files] << file
          routes = opts[:hash_branches][namespace] ||= {}
          meth = routes[segment] = define_roda_method(routes[segment] || "hash_branch_#{namespace}_#{segment}", 1) do |r|
            loc = method(routes[segment]).source_location
            require file
            # Avoid infinite loop in case method is not overridden
            if method(meth).source_location != loc
              send(meth, r)
            end
          end
          nil
        end
```

Reply:
How about this:

Can we eagerly require (as in, regular require) the routes in production, skipping unreloader? This means they can't provide symbols to other files...but that's not what they normally do.

Whereas, as we discovered in loader.rb, converting `.autload` to `require` has implications on dependency tracking between files.